nf_core/components/components_utils.py

```diff
@@ -161,48 +161,39 @@ def get_components_to_install(
                 name, link = match.groups()
                 if link.startswith("../../../"):
                     name_split = name.lower().split("_")
+                    component_name = "/".join(name_split)
                     component_dict = {
-                        "name": "/".join(name_split),
+                        "name": component_name,
                         "org_path": None,
                         "git_remote": None,
                         "branch": None,
                     }
-                    modules.append(component_dict)
+                    modules[component_dict["name"]] = component_dict
                 elif link.startswith("../"):
-                    component_dict = {"name": name.lower(), "org_path": None, "git_remote": None, "branch": None}
-                    subworkflows.append(component_dict)
+                    component_name = name.lower()
+                    component_dict = {"name": component_name, "org_path": None, "git_remote": None, "branch": None}
+                    subworkflows[component_dict["name"]] = component_dict
```

Comment:
It feels like this could be simplified like for the `meta.yml` block below:

1. Keep the `if` statement but only set i) `component_name` and ii) `current_comp_dict` to `subworkflows` or `modules`
2. Then do something like
    ```
    component_dict[component_name] = {
                        "name": component_name,
                        "org_path": None,
                        "git_remote": None,
                        "branch": None,
                    }
    ````